dfusion_rust_core/src/models/util.rs

```diff
@@ -0,0 +1,20 @@
+use std::convert::TryInto;
+use web3::types::U256;
+
+pub fn pop_u8_from_log_data(bytes: &mut Vec<u8>) -> u8 {
+    pop_u256_from_log_data(bytes).as_u32().try_into().unwrap()
```

Comment:
Are we at all worried that popping 256 and only needing 8 wouldn't forget more information than intended?

Reply:
For some reason the data is not compressed, so even though we are only emitting a e.g. u16 from the SC, the event log reserves 32 bytes (u256) for the value. 
These methods are so that fact the developer doesn't have to know about that fact (they should only care how big the type is they are reading/popping).